magefiles/magefile.go

```diff
@@ -1283,7 +1283,7 @@ func CleanWorkload() error {
 func runTests(labelsToRun string, junitReportFile string) error {
 
 	ginkgoArgs := []string{"-p", "--output-interceptor-mode=none", "--no-color",
-		"--timeout=90m", fmt.Sprintf("--output-dir=%s", artifactDir),
+		"--timeout=90m", fmt.Sprintf("--output-dir=%s", artifactDir), "--json-report=e2e-report.json",
```

Comment:
This function is used only by jobs that are not using the rule engine. If you want to apply it to all, I think [this is the right place](https://github.com/konflux-ci/e2e-tests/blob/2cafc4ba2004fca1f0d5f6350ef582301b69f40a/magefiles/rulesengine/repos/common.go#L44)

Reply:
Thank you!